apps/frontend/src/libs/helpers/extract-error-message.helper.ts

```diff
@@ -1,23 +1,53 @@
+import { NotificationMessage, ServerErrorType } from "~/libs/enums/enums.js";
+
+type MaybeHTTPError = {
+	details?: unknown;
+	errorType?: unknown;
+	message?: unknown;
+	name?: unknown;
+	status?: unknown;
+};
+
+const isHttpErrorWithResponse = (
+	error: unknown,
+): error is { response: { data?: { message?: unknown } } } =>
+	typeof error === "object" &&
+	error !== null &&
+	"response" in error &&
+	typeof (error as { response: unknown }).response === "object" &&
+	(error as { response: unknown }).response !== null;
+
+const isCustomHttpError = (error: unknown): error is MaybeHTTPError =>
+	typeof error === "object" &&
+	error !== null &&
+	("errorType" in error || (error as { name?: unknown }).name === "HTTPError");
+
 const extractErrorMessage = (
 	error: unknown,
 	fallbackMessage = "Something went wrong",
 ): string => {
-	if (typeof error === "object" && error !== null) {
-		const maybeResponse = (error as { response?: unknown }).response;
-
-		if (
-			typeof maybeResponse === "object" &&
-			maybeResponse !== null &&
-			"data" in maybeResponse &&
-			typeof (maybeResponse as { data?: unknown }).data === "object"
-		) {
-			const maybeMessage = (maybeResponse as { data: { message?: unknown } })
-				.data.message;
-
-			if (typeof maybeMessage === "string") {
-				return maybeMessage;
-			}
+	if (isCustomHttpError(error)) {
```

Comment:
No need for all that stuff. Before sending a request check if the user is online. If not, throw an error with a custom message